### PR TITLE
doc: add documentation for required parameter

### PR DESCRIPTION
**Pull request by Haswf**
#### What type of PR is this?

This commit adds documentation for required parameter in [Binding and validate](https://www.cloudwego.io/docs/hertz/tutorials/basic-feature/binding-and-validate).
Required paramter is a important binding feature but it's missing in documentation.

#### What this PR does / why we need it (en: English/zh: Chinese):

en: add documentation for required parameter
zh: 补充必填参数文档

### CODE REVIEW

**Comment by Haswf** [comment]
Hey @FGYFFFF, we just discuessed how this features works recently. Could you review this pr?

File: content/en/docs/hertz/tutorials/basic-feature/binding-and-validate.md [hunk 1]
```diff
@@ -57,13 +57,24 @@ func main() {
 
 
 
-### Parameter binding priority
+### Parameter Binding Precedence
```
Comment (Haswf) [inline content/en/docs/hertz/tutorials/basic-feature/binding-and-validate.md:60]: Precedence is a more appropriate word for this context. Also see [Operator precedence
](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence)

File: content/zh/docs/hertz/tutorials/basic-feature/binding-and-validate.md [hunk 2]
```diff
@@ -62,6 +62,17 @@ path > form > query > cookie > header > json > raw_body
 ```
 > 注：如果请求的 content-type 为 `application/json`，那么会在参数绑定前做一次 json unmarshal 处理作为兜底。
 
+### 必传参数
+
+通过在 tag 中添加 `required`，可以将参数标记为必传。当绑定失败时`Bind`和`BindAndValidate`将会返回错误。当多个Tag包含`required`时，将会按照优先级绑定。如果所有tag都没有绑定上，则会返回错误。
```
Comment (GuangmingLuo) [inline content/zh/docs/hertz/tutorials/basic-feature/binding-and-validate.md:67]: tag / Tag 大小写保持一致，注意中英文空格

File: content/zh/docs/hertz/tutorials/basic-feature/binding-and-validate.md [hunk 3]
```diff
@@ -62,6 +62,17 @@ path > form > query > cookie > header > json > raw_body
...
+``` go  
+type TagRequiredReq struct {
+	// 当JSON中没有hertz字段时，默认错误为binding: expr_path=hertz, cause=missing required parameter
```
Comment (GuangmingLuo) [inline content/zh/docs/hertz/tutorials/basic-feature/binding-and-validate.md:70]: 中英空格

**Comment by GuangmingLuo** [comment]
@FGYFFFF please confirm this.

File: content/zh/docs/hertz/tutorials/basic-feature/binding-and-validate.md [hunk 4]
```diff
@@ -62,6 +62,17 @@ path > form > query > cookie > header > json > raw_body
...
 > 注：如果请求的 content-type 为 `application/json`，那么会在参数绑定前做一次 json unmarshal 处理作为兜底。
 
+### 必传参数
+
+通过在 tag 中添加 `required`，可以将参数标记为必传。当绑定失败时 `Bind` 和 `BindAndValidate` 将会返回错误。当多个 tag 包含 `required` 时，将会按照优先级绑定。如果所有 tag 都没有绑定上，则会返回错误。
+``` go  
+type TagRequiredReq struct {
+	// 当 JSON 中没有 hertz 字段时，默认错误为 binding: expr_path=hertz, cause=missing required parameter
+	Hertz string `json:"hertz,required"`
+	Kitex string `query:"kitex,required" json:"kitex,required" `
```
Comment (FGYFFFF) [inline content/zh/docs/hertz/tutorials/basic-feature/binding-and-validate.md:72]: 这个也可以像上面描述下 "// 当 query 和 JSON 中**同时**没有 kitex 字段时，会触发 required 错误：binding: expr_path=hertz, cause=missing required parameter"

File: content/en/docs/hertz/tutorials/basic-feature/binding-and-validate.md [hunk 5]
```diff
@@ -57,13 +57,24 @@ func main() {
...
 
+### Required parameter
+
+You can specify a parameter as required with keyword `required` in tag. Both `Bind` and `BindAndValidate` returns error when a required parameter is missing. 
+When multiple tags contain the`required` keyword, parameter with be bound in order of precedence defined above. If none of the tags bind, an error will be returned.
+``` go  
+type TagRequiredReq struct {
+	// when field hertz is missing in json, default error message is binding: expr_path=hertz, cause=missing required parameter
+	Hertz string `json:"hertz,required"`
+	Kitex string `query:"kitex,required" json:"kitex,required" `
```
Comment (FGYFFFF) [inline content/en/docs/hertz/tutorials/basic-feature/binding-and-validate.md:75]: // when field  kitex is missing in `query` and `JSON` **at the same time**, default error message is binding: expr_path=hertz, cause=missing required parameter

另外上面的 json 字段也可以改成 `JSON`

**Review by FGYFFFF — APPROVED** [review]
lgtm~~~ thanks

**Review by GuangmingLuo — APPROVED** [review]

**Pull request merged**